classes/Api/Firebase/Token.php

```diff
@@ -48,30 +48,48 @@ public function refresh()
         if (isset($response['id_token']) && isset($response['refresh_token'])) {
             \Configuration::updateValue('PS_PSX_FIREBASE_ID_TOKEN', $response['id_token']);
             \Configuration::updateValue('PS_PSX_FIREBASE_REFRESH_TOKEN', $response['refresh_token']);
+            \Configuration::updateValue('PS_PSX_FIREBASE_REFRESH_DATE', date('Y-m-d H:i:s'));
         }
 
         return $response;
     }
 
     /**
-     * Check the token validity. The token expire time is set to 3600 seconds.
+     * Check we can request an other token.
+     *
+     * @return bool
+     */
+    public function shouldRefreshToken()
+    {
+        return $this->hasRefreshToken() && $this->isExpired();
```

Comment:
What if the token does not exist? `$this->hasRefreshToken()` is false, making the whole line return false too.

Reply:
Yes, it’s expected :)
We cannot do a call to request an access_token without a refresh_token even if token is expired.
So if false, refresh() is not called
Currently when you are not logged to Psx or you have logout, this field is empty in Configuration but getToken() call refresh() and make call without request data. As a result it throw an exception due to Bad Request, catch by GenericClient
This fix prevent that call
getToken() will return an empty string and will be considered as not logged like expected